sdk/src/dlob/DLOB.ts

```diff
@@ -1047,6 +1052,33 @@ export class DLOB {
 					bidNode
 				);
 
+				// extra guard against bad fills for perp limit orders where auction is incomplete
+				if (
+					isVariant(takerNode.order.marketType, 'perp') &&
```

Comment:
do we really want to enforce market type here? this seems like a nice check for spot as well?

Reply:
it's a bit tricky to do it on the smart contract, but we always try to fill against serum first, so i think it's ok to remove the perp check here